src/platform/intel/ace/lib/base_fw_platform.c

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2024 Intel Corporation.
+//
+// Author: Kai Vehmanen <kai.vehmanen@linux.intel.com>
+
+#include <rtos/string.h>
+#include <sof/tlv.h>
+#include <sof/lib/dai.h>
+#include <ipc4/base_fw.h>
+
+int platform_basefw_hw_config(uint32_t *data_offset, char *data)
+{
+	struct sof_tlv *tuple = (struct sof_tlv *)data;
+	uint32_t value;
+
+	tlv_value_uint32_set(tuple, IPC4_HP_EBB_COUNT_HW_CFG, PLATFORM_HPSRAM_EBB_COUNT);
```

Comment:
We should put this in ```src/ipc/ipc4/intel.c``` and export several C functions that can be called where vendor data has to be installed (otherwise the are all ```static inline vendor_add_memory()``` and compile out).
The input data for ```IPC4_HP_EBB_COUNT_HW_CFG, PLATFORM_HPSRAM_EBB_COUNT``` has to come from DT, but this could be a secondary PR later on (we could have a comment saying this today).

Reply:
@lgirdwood That's what this patch now does, ipc4/intel.c code in src/platform/intel/ace/lib/base_fw_platform.c and generic stub in src/platform/posix/base_fw_platform.c. 

We can keep this simpler and put this in ipc4/intel.c, but that means we need to maintain a platform/vendor layer in base_fw.h/base_fw_vendor.h and have ifdefs in src/ipc. Current PR keeps the ifdefs out of main src/ipc and ipc/audio trees.

I'm ok with both and can update the PR. There is more code and data to move once the approach is agreed, hw_config is just a start).